File: src/main/java/com/github/nicholashauschild/depinfo/DepInfoPlugin.java
package com.github.nicholashauschild.depinfo;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

/**
 * Author: nicholas.hauschild
 */
public class DepInfoPlugin implements Plugin<Project> {
  static final String EXTENSION_NAME = "dependencyInfo";
  static final String TASK_NAME = "generateDependencyInfo";

  @Override
  public void apply(final Project project) {
    project.getExtensions().create(EXTENSION_NAME, DepInfoExtension.class);
    project.getTasks().create(TASK_NAME, DepInfoTask.class);
  }
}

File: src/main/kotlin/com/github/nicholashauschild/depinfo/DepInfoPlugin.kt
package com.github.nicholashauschild.depinfo

import org.gradle.api.Plugin
import org.gradle.api.Project

val DEP_INFO_TASK_NAME = "generateDependencyInfo"
val DEP_INFO_TASK_CLASS = DepInfoTask::class.java

val EXTENSION_NAME = "dependencyInfo"
val EXTENSION_CLASS = DepInfoExtension::class.java

open class DepInfoPlugin : Plugin<Project> {
    override fun apply(project: Project) {
        project.extensions.create(EXTENSION_NAME, EXTENSION_CLASS)
        project.tasks.create(DEP_INFO_TASK_NAME, DEP_INFO_TASK_CLASS)
    }
}
